python/lsst/dax/ppdb/sql/ppdb_sql_utils.py

```diff
@@ -0,0 +1,293 @@
+        columns = [
+            schema_model.Column(
+                name="apdb_replica_chunk",
+                id=f"#{table_name}.apdb_replica_chunk",
+                datatype=felis.datamodel.DataType.long,
+            ),
+            schema_model.Column(
+                name="last_update_time",
+                id=f"#{table_name}.last_update_time",
+                datatype=felis.datamodel.DataType.timestamp,
+                nullable=False,
+            ),
+            schema_model.Column(
+                name="unique_id",
+                id=f"#{table_name}.unique_id",
+                datatype=schema_model.ExtraDataTypes.UUID,
+                nullable=False,
+            ),
+            schema_model.Column(
+                name="replica_time",
+                id=f"#{table_name}.replica_time",
+                datatype=felis.datamodel.DataType.timestamp,
+                nullable=False,
+            ),
+            schema_model.Column(
+                name="status",
+                id=f"#{table_name}.status",
+                datatype=felis.datamodel.DataType.string,
+                nullable=True,
+            ),
+            schema_model.Column(
+                name="directory",
+                id=f"#{table_name}.directory",
+                datatype=felis.datamodel.DataType.string,
+                nullable=True,
+            ),
+        ]
+        indices = [
+            schema_model.Index(
+                name="PpdbInsertId_idx_last_update_time",
+                id="#PpdbInsertId_idx_last_update_time",
+                columns=[columns[1]],
+            ),
+            schema_model.Index(
+                name="PpdbInsertId_idx_replica_time",
+                id="#PpdbInsertId_idx_replica_time",
+                columns=[columns[3]],
+            ),
+        ]
+
+        # Add table for replication support.
+        chunks_table = schema_model.Table(
+            name=table_name,
+            id=f"#{table_name}",
+            columns=columns,
+            primary_key=[columns[0]],
+            indexes=indices,
+            constraints=[],
+        )
+        return chunks_table
+
+    @classmethod
+    def get_table(cls, sa_metadata: sqlalchemy.schema.MetaData, name: str) -> sqlalchemy.schema.Table:
+        """Get table from SQLAlchemy metadata by name, without including the
+        schema name.
+
+        Parameters
+        ----------
+        sa_metadata : `sqlalchemy.schema.MetaData`
+            Schema definition.
+        name : `str`
+            Name of the table to get.
+
+        Returns
+        -------
+        table : `sqlalchemy.schema.Table`
+            Table object.
+
+        Raises
+        ------
+        LookupError
+            If table with given name does not exist.
+        """
+        for table in sa_metadata.tables.values():
+            if table.name == name:
+                return table
+        raise LookupError(f"Unknown table {name}")
+
+    @classmethod
+    def upsert_replica_chunk(
```

Comment:
This could be made generic and not specific to replica chunks if the ID column name was an argument to the function.